datadog/resource_datadog_dashboard.go

```
@@ -1291,12 +1437,48 @@ func buildTerraformWidget(datadogWidget *datadogV1.Widget) (map[string]interface
+		terraformDefinition := buildTerraformQueryValueDefinition(datadogSourceWidgetDefinition.QueryValueWidgetDefinition)
+		terraformWidgetDefinition["query_value_definition"] = []map[string]interface{}{terraformDefinition}
+	} else if datadogSourceWidgetDefinition.TableWidgetDefinition != nil {
+		terraformDefinition := buildTerraformQueryTableDefinition(datadogSourceWidgetDefinition.TableWidgetDefinition)
+		terraformWidgetDefinition["query_table_definition"] = []map[string]interface{}{terraformDefinition}
+	} else if datadogSourceWidgetDefinition.SunburstWidgetDefinition != nil {
+		terraformDefinition := buildTerraformSunburstDefinition(datadogSourceWidgetDefinition.SunburstWidgetDefinition)
+		terraformWidgetDefinition["sunburst_definition"] = []map[string]interface{}{terraformDefinition}
+	} else if datadogSourceWidgetDefinition.TimeseriesWidgetDefinition != nil {
+		terraformDefinition := buildTerraformTimeseriesDefinition(datadogSourceWidgetDefinition.TimeseriesWidgetDefinition)
+		terraformWidgetDefinition["timeseries_definition"] = []map[string]interface{}{terraformDefinition}
+	} else if datadogSourceWidgetDefinition.ToplistWidgetDefinition != nil {
+		terraformDefinition := buildTerraformToplistDefinition(datadogSourceWidgetDefinition.ToplistWidgetDefinition)
+		terraformWidgetDefinition["toplist_definition"] = []map[string]interface{}{terraformDefinition}
+	} else if datadogSourceWidgetDefinition.TreeMapWidgetDefinition != nil {
+		terraformDefinition := buildTerraformTreemapDefinition(datadogSourceWidgetDefinition.TreeMapWidgetDefinition)
+		terraformWidgetDefinition["treemap_definition"] = []map[string]interface{}{terraformDefinition}
+	} else if datadogSourceWidgetDefinition.GeomapWidgetDefinition != nil {
+		terraformDefinition := buildTerraformGeomapDefinition(datadogSourceWidgetDefinition.GeomapWidgetDefinition)
+		terraformWidgetDefinition["geomap_definition"] = []map[string]interface{}{terraformDefinition}
+	}
```

```suggestion
	} else {
		return nil, fmt.Errorf("unsupported widget type used as split graph source widget: %s", datadogSourceWidgetDefinition.GetActualInstance())
	}
```
same thing as above, if the if/else fails to match it should propagate an error